kubessh/authentication/gitlab.py

```diff
@@ -0,0 +1,40 @@
+from kubessh.authentication import Authenticator
+import async_timeout
+import aiohttp
+import asyncssh
+import re
+from traitlets import Unicode
+
+class GitLabAuthenticator(Authenticator):
+    """
+    Authenticate with GitLab SSH keys
+    """
+    instance_url = Unicode(
+        "",
```

Comment:
Should we set the default to the public GitLab instance?

Reply:
I was wondering about this myself -- without an additional setting that limits the permitted users by default, this would probably be too risky. But we could add a default-empty `allowed_users` like the Github authenticator has